packages/lab-extension/src/interfaceswitcher.tsx

```diff
@@ -0,0 +1,167 @@
+import { IDisposable } from '@lumino/disposable';
+import * as React from 'react';
+
+interface ISwitcherChoice {
+  command: string;
+  dropdownLabel: string;
+  commandLabel: string;
+  urlPrefix: string;
+  current: boolean;
+}
+
+class InterfaceSwitcher extends ReactWidget {
+  private switcherChoices: ISwitcherChoice[];
+  constructor(
+    private commands: CommandRegistry,
+    switcherChoices: ISwitcherChoice[]
+  ) {
+    super();
+    this.addClass('jp-Notebook-toolbarCellType');
+    this.switcherChoices = switcherChoices;
+  }
+
+  render = () => {
+    return (
+      <HTMLSelect
+        className="jp-Notebook-toolbarCellTypeDropdown"
+        value={this.switcherChoices.find(sc => sc.current)?.command}
+      >
+        {this.switcherChoices.map(sc => {
+          return (
+            <option
+              key={sc.command}
+              value={sc.command}
+              onClick={() => !sc.current && this.commands.execute(sc.command)}
+            >
+              {sc.dropdownLabel}
+            </option>
+          );
+        })}
+      </HTMLSelect>
+    );
+  };
+}
+
+/**
+ * A notebook widget extension that adds a open in classic notebook button to the toolbar.
+ */
+class InterfaceSwitcherButton
+  implements DocumentRegistry.IWidgetExtension<NotebookPanel, INotebookModel> {
+  private switcherChoices: ISwitcherChoice[];
+
+  constructor(commands: CommandRegistry, switcherChoices: ISwitcherChoice[]) {
+    this._commands = commands;
+    this.switcherChoices = switcherChoices;
+  }
+
+  createNew(panel: NotebookPanel): IDisposable {
+    const switcher = new InterfaceSwitcher(
+      this._commands,
+      this.switcherChoices
+    );
+    panel.toolbar.insertBefore('kernelName', 'switch-interface', switcher);
+    return switcher;
+  }
+
+  private _commands: CommandRegistry;
+}
+
+/**
+ * A plugin to add custom toolbar items to the notebook page
+ */
+const interfaceSwitcher: JupyterFrontEndPlugin<void> = {
+  id: '@retrolab/lab-extension:interface-switcher',
+  autoStart: true,
+  optional: [INotebookTracker, ICommandPalette, IMainMenu],
+  activate: (
+    app: JupyterFrontEnd,
+    notebookTracker: INotebookTracker | null,
+    palette: ICommandPalette | null,
+    menu: IMainMenu | null
+  ) => {
+    if (!notebookTracker) {
+      // to prevent showing the toolbar button in RetroLab
+      return;
+    }
+
+    const { commands, docRegistry, shell } = app;
+    const baseUrl = PageConfig.getBaseUrl();
+
+    const switcherChoices: ISwitcherChoice[] = [
+      {
+        command: 'retrolab:open-classic',
+        commandLabel: 'Open in Classic Notebook',
+        dropdownLabel: 'Classic',
+        urlPrefix: `${baseUrl}tree/`,
+        current: false
+      },
+      {
+        command: 'retrolab:open-retro',
+        commandLabel: 'Open in RetroLab',
+        dropdownLabel: 'RetroLab',
+        urlPrefix: `${baseUrl}retro/tree/`,
+        current: app.name === 'RetroLab'
```

Comment:
Maybe we could also add `ILabShell` and `IRetroShell` as `optional` dependencies for the plugin, and check whether they are `null` to know if the current interface is a JupyterLab-like or RetroLab-like interface?

Reply:
`name` could indeed be overriden via the page config or directly in JS like here: 

https://github.com/jtpio/jupyterlite/blob/a198d902908483f8eea36fecae858cbd14b4a078/app/lab/index.template.js#L176